proxy/run.go

```diff
@@ -184,6 +184,7 @@ func Run(ctx context.Context, args []string) int {
 		Tokens:            cfg.Tokens,
 		Peers:             cfg.Peers,
 		IdempotentGraph:   cfg.IdempotentGraph,
+		ConnectTimeout:    cfg.AstraTimeout,
```

Comment:
These feel like two different kinds of timeouts.  AstraTimeout currently governs the timeout for actually retrieving metadata or loading the bundle from Astra, both of which are very explicitly related to Astra.  Changing this to also be a general connect timeout feels like a mixing of concerns.

I very much like the idea of being able to explicitly specify a connect timeout, however.  Maybe we add that as an explicit CLI flag, perhaps using Astra timeout if that's specified but the connect timeout isn't?  i.e. connect timeout = connect timeout || astra timeout || some default value?